include/roboteam_ai/control/positionControl/BBTrajectories/WorldObjects.h

```diff
@@ -0,0 +1,123 @@
+                                            const std::vector<Vector2> &pathPoints, int robotId, double timeStep);
+
+        /**
+         * @brief Takes a calculated path of a robot and checks points along the path if they are too close to an
+         * approximation of the ball trajactory. If the play is "ball_placement_them" also checks for the path
+         * being inside the balltube. Adds these points and the time to collisionDatas and collisionTimes
+         * @param collisionDatas, std::vector which rtt::BB::CollisionData can be added to
+         * @param collisionTimes, std::vector which times can be added to
+         * @param pathPoints, std::vector with path points
+         * @param timeStep in seconds
+         */
+        void calculateBallCollisions(std::vector<CollisionData> &collisionDatas, std::vector<double> &collisionTimes,
+                                     std::vector<Vector2> pathPoints, double timeStep);
+
+        /**
+         * @brief Takes a calculated path of a robot and checks points along the path if they are too close to an
+         * approximation of the enemy robot paths. Adds these points and the time to collisionDatas and collisionTimes if
+         * the difference in velocity between the two robots is more than 1.5 ms/s and we are driving faster
+         * @param BBTrajectory BBTrajectory which is used for getting the velocity of the robot
+         * @param collisionDatas, std::vector which rtt::BB::CollisionData can be added to
+         * @param collisionTimes, std::vector which times can be added to
+         * @param pathPoints, std::vector with path points
+         * @param timeStep in seconds
+         */
+        void calculateEnemyRobotCollisions(rtt::BB::BBTrajectory2D BBTrajectory, std::vector<CollisionData> &collisionDatas,
+                                           std::vector<double> &collisionTimes, const std::vector<Vector2> &pathPoints,
+                                           double timeStep);
+
+        /**
+         * @brief Takes a path from the array of stored paths and checks points along the path if they are too close to
+         * where our robots are calculated to be at that point in time. Adds these points and the time to collisionDatas
+         * and collisionTimes
+         * @param collisionDatas, std::vector which rtt::BB::CollisionData can be added to
+         * @param collisionTimes, std::vector which times can be added to
+         * @param pathPoints, std::vector with path points
+         * @param robotId
+         * @param timeStep in seconds
+         */
+        void calculateOurRobotCollisions(std::vector<CollisionData> &collisionDatas, std::vector<double> &collisionTimes,
+                                         const std::vector<Vector2> &pathPoints, int robotId, double timeStep);
+
+        void setField(const rtt::ai::rtt_world::Field &field);
```

Comment:
This seems dangerous, as you have a pointer field; it's probably better to pass world and field pointers and/or/references to every single function call.